cns/requestcontroller/kubecontroller/crdreconciler.go

```diff
@@ -0,0 +1,53 @@
+package kubecontroller
+
+import (
+	"context"
+
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/requestcontroller"
+	nnc "github.com/Azure/azure-container-networking/nodenetworkconfig/api/v1alpha"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// CrdReconciler watches for CRD status changes
+type CrdReconciler struct {
+	APIClient APIClient
+	NodeName  string
+	CNSClient requestcontroller.CNSClient
+}
+
+// Reconcile is called on CRD status changes
+func (r *CrdReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	var nodeNetConfig nnc.NodeNetworkConfig
+
+	//Get the CRD object
+	if err := r.APIClient.Get(context.TODO(), request.NamespacedName, &nodeNetConfig); err != nil {
+		logger.Printf("[cns-rc] CRD not found, ignoring %v", err)
```

Comment:
There are other reasons as to why Get could return an error. Instead of this logging message, I would change the logic to something like
```go
if err := r.APIClient.Get(context.TODO(), request.NamespacedName, &nodeNetConfig); err != nil {
    if apierrors.IsNotFound(err) {
        logger.Printf("[cns-rc] CRD not found in cache, ignoring %v", err)
        return reconcile.Result{}, nil
    } else {
        logger.Printf("[cns-rc] error retrieving CRD %v from cache", err)
        return reconcile.Result{}, err
    }
}
```